### ISSUE TITLE
No problems reported using a custom .groovylintrc.json referencing existing codenarcRulesets

### ISSUE THREAD

**Issue by fun4jimmy** [nvuillam/vscode-groovy-lint#91]
If I run codenarc using the xml ruleset configurations I have defined I get a lot of rule violations. Looking at the same files in vscode with the vscode-groovy-lint extension enabled there are no problems reported (eg. "No problems have been detected in the workspace so far.").

My project layout is:
- config
  - codenarc
    - codenarc_main.xml
    - codenarc_test.xml
- src
  - com
    - company
      - some helper class .groovy files
- test
  -  .groovylintrc.json
 - vars
   - jenkins variable .groovy files
- .groovylintrc.json

**config\codenarc\codenarc_main.xml:**
```xml
<ruleset xmlns="http://codenarc.org/ruleset/1.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://codenarc.org/ruleset/1.0 http://codenarc.org/ruleset-schema.xsd"
         xsi:noNamespaceSchemaLocation="http://codenarc.org/ruleset-schema.xsd">
    <ruleset-ref path='rulesets/basic.xml'/>
    <!-- other rulesets -->
</ruleset>
````
**config\codenarc\codenarc_test.xml:**
```xml
<ruleset xmlns="http://codenarc.org/ruleset/1.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://codenarc.org/ruleset/1.0 http://codenarc.org/ruleset-schema.xsd"
         xsi:noNamespaceSchemaLocation="http://codenarc.org/ruleset-schema.xsd">
    <ruleset-ref path='file:config/codenarc/codenarc_main.xml'>
        <!-- Some excludes for rules adding by codenarc_main. -->
    </ruleset-ref>
    <!-- other rulesets -->
</ruleset>
````
**test\\.groovylintrc.json:**
```json
{
  "codenarcRulesets": "../config/codenarc/codenarc_test.xml"
}
````
**.groovylintrc.json:**
```json
{
  "codenarcRulesets": "config/codenarc/codenarc_main.xml"
}
````

The vscode output from Groovy Lint contains errors:
```
Start analyzing file:///e%3A/omg/jenkins/jenkins-pipeline-vicon/vars/skippableStage.groovy
GroovyLint: Started CodeNarc Server
Completed analyzing file:///e%3A/omg/jenkins/jenkins-pipeline-vicon/vars/skippableStage.groovy in 7031 ms
(node:13060) UnhandledPromiseRejectionWarning: Error: getaddrinfo ENOENT api.amplitude.com
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:60:26)
(node:13060) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:13060) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

The extension was initially working and highlighting problems when I had no configuration files setup.

**Comment by fun4jimmy** [comment]
I have done a bit more debugging today after enabling the Visual Studio Code extension "Groovy Lint > Debug: Enable" setting.

In the log it appears to generate a temporary ruleset file which contains no reference to my XML ruleset.

> 2020-11-02T17:48:59.841Z npm-groovy-lint CREATE RULESET tmp file C:\Users\<USERNAME>\AppData\Local\Temp\npm-groovy-lint\codeNarcTmpRs_0.9198010039621833.groovy generated from input options, as CodeNarc requires physical files

**codeNarcTmpRs_0.9198010039621833.groovy:**
```groovy
ruleset {

    description 'Generated by npm-groovy-lint (https://github.com/nvuillam/npm-groovy-lint#readme)'

    Indentation(enabled:false)

}
```

Is there some code missing to handle the codenarcRulesets property from the GroovyLint config file?

**Comment by luiszimmermann** [comment]
Same thing here, it appears that never worked, using an old extension version procudes the same error. Using `.xml` or `.groovy` in the `codenarcRulesets` does nothing.

**Comment by nvuillam** [comment]
i'll try to have a look during these vacations, ,thanks for reporting
If someone wants to debug it, it is also very welcome :)

**Comment by mbudnek** [comment]
I just ran into this and dug into it a bit.  It looks like this extension is passing its indentation config to the underlying npm-groovy-lint package using its `rulesets` option.  Since that option has been manually passed in, it overrides the option loaded from the `codenarcRulesets` config file option.  Even if it were to merge the manually-set rule with the ruleset from the config file, `buildRuleSets` mangles the manual rule by converting it to a `file://` URI.

**Comment by nvuillam** [comment]
Yes I confirm the indentation is passed, to avoid to have to define it manually (I had to do that after another issue in the past, because users were unhappy about the 4 spaces default indentation)
Do you have a resolution proposition ? (I'm sorry I started a new job and have other projects, It's had to find free time to work on VsCodeGroovyLint these times :/ )

**Comment by viktoria7211** [comment]
I run into the same problem - no linter issues found if setting `codenarcRulesets`.
 
@nvuillam are there already any updates on this issue? :)
